torch/_inductor/codegen/cpp.py

```diff
@@ -83,21 +83,29 @@ def reduction_init(reduction_type, dtype):
     raise AssertionError(reduction_type)
 
 
-def reduction_combine(reduction_type, var, next_value):
+def reduction_combine(reduction_type, var, next_value, threads_num=0):
     if reduction_type == "sum":
-        return f"{var} += {next_value}"
+        if threads_num == 0:
+            return f"{var} += {next_value}"
+        else:
+            var_with_id = var.replace(f"[{threads_num}]", "[tid]")
```

Comment:
The number of threads could be dynamic. Better we always get the number from OMP runtime.